Refactor: codebase & documentation

lemuelmgsn opened the pull request:
## What does this change?

References issue #337 , #344 
Closes issue #338 , #339 , #340 

This PR updates the documentation and refactors a couple of +page.svelte pages using components. 
In this PR i refactored half of all the +page.svelte pages. I will refactor all the other +page.svelte pages in an other branch

These are the pages i refactored:

- all stories page
- choose buddy page
- languages page
- lessons page


<!-- A PR should have enough detail to be understandable far in the future. e.g what is the problem/why is the change needed, how does it solve it and any questions or points of discussion. Prefer copying information from a GitHub issue over linking to it; the card may not always exist and reviewers may not have access to the board. -->

[livesite](https://livesite.com)


## Images

<!-- Usually only applicable to UI changes, what did it look like before and what will it look like after? -->

<img width="743" height="736" alt="image" src="https://github.com/user-attachments/assets/aaee97a0-d364-43aa-9ee5-79de3263aaa2" />



## How to review

<!-- Provide instructions to help others verify the change. This could take the form of "On PROD, do X and witness Y. On this branch, do X and witness Z. " -->
- Verify if all names are correct
- if all pages work like they are supposed to

lemuelmgsn committed:
Merge pull request #336 from fdnd-agency/dev

dev into main after reviewed by @marjamlodien & @roumaisaHVA

lemuelmgsn committed:
Update README.md

lemuelmgsn committed:
refactor & doc: removed unused files and updated contributing.md

closes #340 , #338

lemuelmgsn committed:
docs: update technische-docs.md

closes #341
references #337

lemuelmgsn committed:
refactor: all-stories page with components

references #344

lemuelmgsn committed:
fix: better names for filter component

lemuelmgsn committed:
refactor: choose buddy page into componets

lemuelmgsn committed:
refactor: languages page with components

references #344

lemuelmgsn committed:
refactor: lessons page with components

references #344

lemuelmgsn committed:
refactor & fix: refactored log-in page and fixed styling

turned sections into components and fixed styling issues.

resolves #346

Jesse-L3 commented on src/lib/components/core/BuddyCarousel.svelte:
```diff
@@ -0,0 +1,170 @@
+  .scroll-container {
+    min-height: 26.75em;
+    width: 100%;
+    max-width: 720px;
```
Reccomend to work with em for things like this

lemuelmgsn replied:
Changed it to em

Jesse-L3 commented on src/lib/components/core/BuddyCarousel.svelte:
```diff
@@ -0,0 +1,170 @@
+    flex: 0 0 auto;
+    text-align: center;
+    padding: 0.3em;
+    border-radius: 10px;
```
smaller things like this can stay px

Jesse-L3 commented on src/lib/components/forms/ContinueBtn.svelte:
```diff
@@ -12,43 +12,43 @@
-    --button-darker-blue: hsl(200, 49%, 26%);
-    --button-grey: hsl(0, 0%, 75%);
-  }
+:global(:root) {
```
Inst it better to use this in your global css if this is a global root style?

OniWithTheHoodie replied:
i agree with Jesse look at all your components that have a root within the style tag. place them in a gloabal.css file and then call the custom properties that you need. will also make your component even smaller which is pleasent to read for fellow developers that will continue with your project.

lemuelmgsn replied:
yes i forgot to delete them in the +page.svelte, thanks

Jesse-L3 commented on src/lib/index.js:
```diff
@@ -3,6 +3,23 @@ export { default as MakePlaylist } from '$lib/components/core/MakePlaylist.svelt
 export { default as Menu } from '$lib/components/core/Menu.svelte'
```
Nice way to categories you components

Jesse-L3 commented on src/routes/lessons/+page.svelte:
```diff
@@ -1,271 +1,69 @@
+  }
 
-@media only screen and (min-width: 600px) {
+  @media only screen and (min-width: 600px) {
```
remove the `only screen and`

OniWithTheHoodie replied:
what Jesse said

Jesse-L3 commented on static/style.css:
reccomend to rename this file to global.css

Jesse-L3 reviewed (changes requested):
Overall it looks pretty good. Your code looks clean and nice. and its well orginized. Make sure to fix the feedback that i gave you and then youre good to go.

Dont forget to make every page.svelte, component only

OniWithTheHoodie commented on src/lib/components/core/LoginIntro.svelte:
```diff
@@ -0,0 +1,91 @@
+
+<style>
+
+    :root {
```
root not aligned with the rest of you css code stay consitent for readability also wouldnt it be better to place this in a global.css file ?

lemuelmgsn replied:
deleted it!

OniWithTheHoodie commented on src/lib/components/core/LoginIntro.svelte:
```diff
@@ -0,0 +1,91 @@
+    --popup-width: 25em;
+    --popup-height: 31.25em
+}
+  section {
```
section not aligned with the rest of you css code stay consitent for readability

OniWithTheHoodie commented on src/lib/components/core/LoginIntro.svelte:
```diff
@@ -0,0 +1,91 @@
+  p {
+    font-size: var(--font-size-md);
+  }
+  h1, p{
```
give it a enter for some breathing space

OniWithTheHoodie commented on src/lib/components/core/LoginIntro.svelte:
```diff
@@ -0,0 +1,91 @@
+  h1, p{
+    color: white;
+    text-shadow: 0 2px 4px rgba(0, 0, 0, 0.4); 
+  }
```
enter here as well

OniWithTheHoodie commented on src/lib/components/core/LoginIntro.svelte:
```diff
@@ -0,0 +1,91 @@
+    color: var(--color-white);
+    text-align: center;
+    cursor: pointer;
+  }
```
enter here as well

OniWithTheHoodie commented on src/lib/components/core/LoginIntro.svelte:
```diff
@@ -0,0 +1,91 @@
+    text-decoration: underline;
+    max-width: 22rem;
+    text-align: center;
+    
```
close off this empty spacing

OniWithTheHoodie commented on src/lib/components/core/LoginPopup.svelte:
```diff
@@ -0,0 +1,220 @@
+
+<style>
+
+:root {
```
i saw another root before in LoginTinro component put them together in a global.css file

OniWithTheHoodie commented on src/lib/components/core/LoginPopup.svelte:
```diff
@@ -0,0 +1,220 @@
+  top: 150%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+
```
close of empty space for better readability

OniWithTheHoodie commented on src/lib/components/core/LoginPopup.svelte:
```diff
@@ -0,0 +1,220 @@
+
+  width: var(--popup-width);
+  height: var(--popup-height);
+
```
close of empty space for better readability

OniWithTheHoodie commented on src/lib/components/core/LoginPopup.svelte:
```diff
@@ -0,0 +1,220 @@
+  border-radius: 1.25em;
+  padding: 1.25em;
+  box-sizing: border-box;
+
```
close of empty space for better readability

OniWithTheHoodie commented on src/lib/components/core/LoginPopup.svelte:
```diff
@@ -0,0 +1,220 @@
+.login-popup {
+  width: 100%;
+  padding: 0.8rem;
+
```
close of empty space for better readability

OniWithTheHoodie commented on src/lib/components/core/LoginPopup.svelte:
```diff
@@ -0,0 +1,220 @@
+  color: var(--color-white);
+  border: 1px solid var(--color-white);
+  border-radius: 0.6em;
+
```
close of empty space for better readability

OniWithTheHoodie commented on src/routes/lessons/+page.svelte:
```diff
@@ -1,271 +1,69 @@
 <style>
-
-:root {
+  :root {
```
maybe add this to your global.css if it's for the whole website ? or is it specific to a component or page ?

OniWithTheHoodie reviewed (changes requested):
it's looking good just have to check the feedback that you have received to make the code even better. you're doing well, keep it up :)

lemuelmgsn committed:
refactor: new buddy component and px into em

lemuelmgsn committed:
refactor: processed feedback

processed feedback by fixing spacing, fixing global.css and removing unnecesary code

lemuelmgsn committed:
Merge branch 'dev' into 337-refactor-refactor-codebase
